mobile/native/src/trade/position/handler.rs

```diff
@@ -39,34 +39,47 @@ pub async fn trade(filled: FilledWith) -> Result<()> {
         filled_with: filled,
     };
 
-    let execution_price = trade_params
+    // If we have a position _and_ the order is not closing the position (i.e. the contracts between
+    // position and order do not match), we must be resizing the position.
+
+    let trade_execution_price = trade_params
         .average_execution_price()
         .to_f32()
         .expect("to fit into f32");
-    order::handler::order_filling(order.id, execution_price)
-        .context("Could not update order to filling")?;
-
-    // If we have a position _and_ the order is not closing the position (i.e. the contracts between
-    // position and order do not match), we must be resizing the position.
 
-    if let Some(Position {
-        quantity,
-        position_state: PositionState::Open,
-        direction,
-        contract_symbol,
-        ..
-    }) = get_positions()?.first()
-    {
-        let position_contracts_relative = compute_relative_contracts(*quantity, *direction);
-        let trade_contracts_relative =
-            compute_relative_contracts(trade_params.quantity, trade_params.direction);
-
-        if *contract_symbol == trade_params.contract_symbol
-            && position_contracts_relative + trade_contracts_relative != Decimal::ZERO
-        {
-            set_position_state(PositionState::Resizing)?;
+    let execution_price = match get_positions()?.first() {
```

Comment:
🤔 I don't get why we need to compute an execution price. Surely, the execution price should be the one from the trade params, no matter the state of the current position. Examples:

1. If we are 5 long and we go 5 long at price X, we are now 10 long with an average execution price that takes into account price X.
2. If we are 5 long and we go 2 short, we are now 3 long and the PNL uses the trade execution price as a closing price. Importantly the average execution price for the position should not change in such a scenario. I need to check that.
3. If we are 5 long and we got 5 short, we no longer have a position and the PNL uses the trade execution price as the closing price.
4. If we are 5 long and we go 10 short, we are now 5 short with the trade execution price as the new average entry price. The PNL for the closed long position is calculated using the trade execution price as the closing price.

I don't think we want to mess with the trade execution price we record in the database in any of these scenarios.

Reply:
> I need to check that.

I found one UI bug already. We were wrongly updating the average execution price of the position when partially reducing the position.